src/async_wrap.cc

```diff
@@ -301,11 +301,11 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
   CHECK(name##_v->IsFunction());                                              \
   env->set_async_hooks_##name##_function(name##_v.As<Function>());
 
-  SET_HOOK_FN(init);
-  SET_HOOK_FN(before);
-  SET_HOOK_FN(after);
-  SET_HOOK_FN(destroy);
-  SET_HOOK_FN(promise_resolve);
+  SET_HOOK_FN(init)
+  SET_HOOK_FN(before)
+  SET_HOOK_FN(after)
+  SET_HOOK_FN(destroy)
+  SET_HOOK_FN(promise_resolve)
```

Comment:
Can you convert `SET_HOOK_FN` to use `do { ... } while(0)` instead, which is the more customary variant for macros which expand to groups of statements? (That also allows us to get rid of the `name##_v` hack inside the macro, it can just be `name`)

Reply:
@addaleax Can you review this again ? Not quite sure I got this right.